main.rb

```diff
@@ -0,0 +1,20 @@
+require_relative './enumerables'
+# Use MyEnumerable module on initialized variable
+class MyList
+  include MyEnumerable
+  def initialize(*list)
+    @list = list.flatten
+  end
+
+  def each(&block)
+    @list.each(&block)
+    self
+  end
```

Comment:
- [ ] I would recommend writing `each` method in your own way without using ruby `each` predefine method.

```ruby
def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
```

Reply:
Thank you. We just implemented this change. Please can you explain how the module works with the class? We don't still seem to understand how "each" calls the all, any and filter methods in the module in the other file.

Reply:
This issue is well resolved! ✔️